[apiConformance] Add tests on props

sbalandi opened the pull request:
### Details:
 - *item1*
 - *...*

### Tickets:
 - *CVS98924*

iefode commented on src/tests/functional/plugin/conformance/test_runner/api_conformance_runner/src/behavior/ov_plugin/properties.cpp:
```diff
@@ -54,4 +54,52 @@ INSTANTIATE_TEST_SUITE_P(ov_plugin_AutoBatch, OVPropertiesTests,
+        ::testing::Combine(
+                        ::testing::Values(CommonTestUtils::DEVICE_HETERO, CommonTestUtils::DEVICE_AUTO, CommonTestUtils::DEVICE_MULTI),
+                        ::testing::ValuesIn(OVCheckChangePropComplieModleGetPropTests::getPropertiesValues()),
+                        ::testing::ValuesIn(ov::test::conformance::generate_ov_configs(CommonTestUtils::DEVICE_AUTO, {}))),
```
why '::testing::ValuesIn(ov::test::conformance::generate_ov_configs(CommonTestUtils::DEVICE_AUTO, {}))),'??

iefode commented on src/tests/functional/plugin/conformance/test_runner/api_conformance_runner/src/behavior/ov_plugin/properties.cpp:
```diff
@@ -54,4 +54,52 @@ INSTANTIATE_TEST_SUITE_P(ov_plugin_AutoBatch, OVPropertiesTests,
+        ::testing::Combine(
+                        ::testing::Values(CommonTestUtils::DEVICE_HETERO, CommonTestUtils::DEVICE_AUTO, CommonTestUtils::DEVICE_MULTI),
+                        ::testing::ValuesIn(device_properties),
+                        ::testing::ValuesIn(ov::test::conformance::generate_ov_configs(CommonTestUtils::DEVICE_AUTO, {}))),
```
the same

sbalandi replied:
because it is no difference in behavior of function for HETERO\MULTI\AUTO and I can put any

ilya-lavrenov commented on src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp:
```diff
@@ -234,6 +235,144 @@ TEST_P(OVClassExecutableNetworkGetMetricTest_EXEC_DEVICES, CanGetExecutionDevice
+    }
+
+    res.push_back({{ov::PropertyName(ov::inference_num_threads.name(), ov::inference_num_threads.mutability), 1}});
+    res.push_back({{ov::PropertyName(ov::compilation_num_threads.name(), ov::compilation_num_threads.mutability), 1}});
```
yes

sbalandi replied:
@ilya-lavrenov  but they have same description and alias and type of value, it is okay ? what is the difference between them ?

```
/**
 * @brief The number of executor logical partitions
 * @ingroup ov_runtime_cpp_prop_api
 */
static constexpr Property<Num, PropertyMutability::RW> num{"NUM_STREAMS"};
```

```
/**
 * @brief The number of executor logical partitions
 * @ingroup ov_runtime_cpp_prop_api
 */
static constexpr Property<streams::Num, PropertyMutability::RW> num_streams{"NUM_STREAMS"};
```

iefode commented:
@sbalandi Please fix my comments, I will merge

sbalandi commented:
> @sbalandi Please fix my comments, I will merge



> @sbalandi Please fix my comments, I will merge

Hi, I fixed, also, I add test OVCheckChangePropComplieModleGetPropTests_ModelDependceProps because ov::optimal_batch_size and ov::max_batch_size is need of ov::hint::model , so i have moved it to another test
